ddtrace/appsec/_ai_guard/_langchain.py

```diff
@@ -0,0 +1,179 @@
+from functools import singledispatch
+from typing import Optional
+
+from langchain_core.agents import AgentAction
+from langchain_core.agents import AgentFinish
+from langchain_core.messages.ai import AIMessage
+from langchain_core.messages.base import BaseMessage
+from langchain_core.messages.function import FunctionMessage
+from langchain_core.messages.human import HumanMessage
+from langchain_core.messages.system import SystemMessage
+from langchain_core.messages.tool import ToolMessage
+
+from ddtrace.appsec.ai_guard import AIGuardAbortError
+from ddtrace.appsec.ai_guard import AIGuardClient
+from ddtrace.appsec.ai_guard import Prompt
+from ddtrace.appsec.ai_guard import ToolCall
+from ddtrace.appsec.ai_guard._api_client import Evaluation
+from ddtrace.contrib.internal.langchain.patch import DispatchResult
+from ddtrace.contrib.internal.trace_utils import unwrap
+from ddtrace.contrib.internal.trace_utils import wrap
+import ddtrace.internal.logger as ddlogger
+
+
+logger = ddlogger.get_logger(__name__)
+
+client: AIGuardClient
+
+
+def langchain_listen(core, ai_guard: AIGuardClient):
```

Comment:
I think the langchain_listen function makes more sense in ddtrace/appsec/_ai_guard/_listener.py.